src/Services/Authentication/UserLoginService.php

```diff
@@ -21,14 +22,33 @@ public function __construct(
     /**
      * @throws AuthenticationException
      */
-    public function loginUser(string $email, string $password): void
+    public function checkUser(string $email, string $password): string
     {
-        $user = User::where("email", $email)->first();
-
+        $user = User::query()->where("email", $email)->first();
         if (!$this->hasher->check($password, $user?->password)) {
             throw new AuthenticationException("Bad credentials");
         }
+        if ($user->isTwoFaEnable) {
+            $this->generateCode($user);
+            return "TwoFa.index";
+        }
+        $this->loginUser($user);
+        return "home";
+    }
+
+    public function generateCode(User $user): void
+    {
+        $code = rand(100000, 999999);
+
+        $user->code()->updateOrCreate(
+            ["user_id" => $user->id],
+            ["code" => $code],
+        );
```

Comment:
```suggestion
        $user->code()->updateOrCreate(
            "user_id" => $user->id,
            "code" => $code,
        );
```
Why arrays in array?

Reply:
i am using two parameters. First one is a condition for a match and the second one specify which fields update.
If I only used one parameter like you suggest the condition will never match and will create new data instead of updating.